catalog/spatial/registry/registry-report-viewer/src/main/java/org.codice.ddf.registry.report.viewer/RegistryReportViewer.java

```diff
@@ -0,0 +1,486 @@
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryObjectType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryPackageType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ServiceBindingType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ServiceType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotType1;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.TelephoneNumberType;
+
+/**
+ * Generates a registry report for a given registry metacard.
+ * Displays the Name, General, Collections,
+ * Services, Organizations, and Contacts information for the node.
+ */
+@Path("/")
+public class RegistryReportViewer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegistryReportViewer.class);
+
+    private static final String BINDING_TYPE = "bindingType";
+
+    private static final String NAME = "Name";
+
+    private static final String VERSION_INFO = "Version Info";
+
+    private static final String DESCRIPTION = "Description";
+
+    private static final String GENERAL = "General";
+
+    private static final String COLLECTIONS = "Collections";
+
+    private static final String REGISTRY_NAME = "RegistryName";
+
+    private static final String SERVICES = "Services";
+
+    private static final String ORGANIZATIONS = "Organizations";
+
+    private static final String CONTACTS = "Contacts";
+
+    private static final String ACCESS_URL = "Access Url";
+
+    private static final String SERVICE = "Service";
+
+    private static final String TARGET_BINDING = "Target binding";
+
+    private static final String SPECIFICATION_LINK = "Specification link";
+
+    private static final String PROPERTIES = "Properties";
+
+    private static final String BINDINGS = "Bindings";
+
+    private static final String ADDRESSES = "Addresses";
+
+    private static final String PHONE_NUMBERS = "Phone Numbers";
+
+    private static final String EMAIL_ADDRESSES = "Email Addresses";
+
+    private static final String CONTACT_INFO = "ContactInfo";
+
+    private static final String CUSTOM_SLOTS = "CustomSlots";
+
+    private FederationAdminService federationAdminService;
+
+    private ClassPathTemplateLoader templateLoader;
+
+    private InternationalStringTypeHelper internationalStringTypeHelper =
+            new InternationalStringTypeHelper();
+
+    private RegistryPackageTypeHelper registryPackageTypeHelper = new RegistryPackageTypeHelper();
+
+    private SlotTypeHelper slotTypeHelper = new SlotTypeHelper();
+
+    public RegistryReportViewer() {
+        templateLoader = new ClassPathTemplateLoader();
+        templateLoader.setPrefix("/templates");
+        templateLoader.setSuffix(".hbt");
+    }
+
+    @Path("/{registryId}/report")
+    @Produces(MediaType.TEXT_HTML)
+    @GET
+    public Response viewRegInfoHtml(@PathParam("registryId") final String registryId,
+            @QueryParam("sourceId") final List<String> sourceIds) {
+        Handlebars handlebars = new Handlebars(templateLoader);
```

Comment:
I don't know how expensive it is to create this object, but you probably shouldn't bother doing so until closer to when you need it. At least after this first validation.